docs/how-to/hip_runtime_api/memory_management/unified_memory.rst

```diff
@@ -0,0 +1,737 @@
+while still keeping the benefits of the dedicated physical memories, by
+supporting the concept of unified memory. This enables the CPU and the GPUs in
+the system to access host and other GPUs' memory without explicit memory
+management.
+
+Unified memory
+================================================================================
+
+Unified Memory is a single memory address space accessible from any processor
+within a system. This setup simplifies memory management and enables
+applications to allocate data that can be read or written on both CPUs and GPUs
+without explicitly copying it to the specific CPU or GPU. The Unified memory
+model is shown in the following figure.
+
+.. figure:: ../../../data/how-to/hip_runtime_api/memory_management/unified_memory/um.svg
+
+Unified memory enables the access to memory located on other devices via
+several methods, depending on whether hardware support is available or has to be
+managed by the driver.
+
+Hardware supported on-demand page migration
+--------------------------------------------------------------------------------
+
+When a kernel on the device tries to access a memory address that is not in its
+memory, a page-fault is triggered. The GPU then in turn requests the page from
+the host or an other device, on which the memory is located. The page is then
+unmapped from the source, sent to the device and mapped to the device's memory.
+The requested memory is then available to the processes running on the device.
+
+In case the device's memory is at capacity, a page is unmapped from the device's
+memory first and sent and mapped to host memory. This enables more memory to be
+allocated and used for a GPU, than the GPU itself has physically available.
+
+This level of unified memory support can be very beneficial for sparse accesses
+to an array, that is not often used on the device.
+
+Driver managed page migration
+--------------------------------------------------------------------------------
+
+If the hardware does not support on-demand page migration, then all the pages
+accessed by a kernel have to be resident on the device, so they have to be
+migrated before the kernel is running. Since the driver can not know beforehand,
+what parts of an array are going to be accessed, all pages of all accessed
+arrays have to be migrated. This can lead to significant delays on the first run
+of a kernel, on top of possibly copying more memory than is actually accessed by
+the kernel.
+
+.. _unified memory system requirements:
+
+System requirements
+================================================================================
+
+Unified memory is supported on Linux by all modern AMD GPUs from the Vega
+series onward. Unified memory management can be achieved with managed memory
+allocation and, for the latest GPUs, with a system allocator.
+
+The table below lists the supported allocators for the different architectures.
```

Comment:
I think you can eliminate this paragraph after noting it in the prior paragraph.

Reply:
Done